src/components/Table/TableHead/TableHead.jsx

```diff
@@ -350,14 +354,44 @@ const TableHead = ({
                 'table-header-sortable': matchingColumnMeta.isSortable,
                 [`${iotPrefix}--table-header-resize`]: hasResize,
               })}
+              // data-floating-menu-container is a work around for this carbon issue: https://github.com/carbon-design-system/carbon/issues/4755
+              data-floating-menu-container
             >
               <TableCellRenderer
+                className={`${iotPrefix}--table-head--text`}
                 wrapText={wrapCellText}
                 truncateCellText={truncateCellText}
                 allowTooltip={false}
               >
                 {matchingColumnMeta.name}
               </TableCellRenderer>
+
+              {matchingColumnMeta.options !== undefined && matchingColumnMeta.options !== null ? (
+                <OverflowMenu
+                  className={`${iotPrefix}--table-head--overflow`}
+                  direction="bottom"
+                  data-testid="table-head--overflow"
+                  flipped={columnIndex !== 0}
+                  onClick={e => {
+                    e.stopPropagation();
+                  }}
+                >
+                  {matchingColumnMeta.options.map((option, index) => (
+                    <OverflowMenuItem
+                      itemText={option.text}
+                      primaryFocus={index === 0}
+                      key={`${columnIndex}--overflow-item-${index}`}
+                      onClick={e => {
+                        e.stopPropagation();
```

Comment:
let's define this onClick function outside of the mapping function and pass the event in
```
onClick={e => handleOnClick(e)}
```